### ISSUE TITLE
Update query for monitored function

### ISSUE THREAD

**Issue by gomoripeti** [Appliscale/xprof#90]
Allow to update the query (in place in the graph title bar) for already monitored functions. This can be done without backend modification with a demonitor/monitor combo.
Some restrictions may apply such as only allow to modify the match-spec part and not module and function.
Some attention needs to be payed for error handling, eg if the new query is invalid keep the old graph unchanged.

**Comment by afronski** [comment]
In such case I think without additional marker on graph it may introduce confusion (e.g. vertical line that marked change in the query with before / after).

**Comment by gomoripeti** [comment]
True, I just had in mind (laziness) not to reenter the query and just clear the graph all together.
But good idea to keep old results but with a very much visible mark that from a certain point we are monitoring something completely different.
